govuk-notify-spi/src/main/scala/uk/gov/nationalarchives/notifyspi/NotifyEmailSenderProvider.scala

```diff
@@ -0,0 +1,44 @@
+package uk.gov.nationalarchives.notifyspi
+
+import java.util
+
+import org.keycloak.email.EmailSenderProvider
+import org.keycloak.models.UserModel
+import uk.gov.service.notify.{NotificationClient, NotificationClientException}
+
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+class NotifyEmailSenderProvider(notifyClient: NotificationClient) extends EmailSenderProvider {
+
+  private val templateIdKey = "GOVUK_NOTIFY_TEMPLATE_ID"
+
+  override def send(config: util.Map[String, String],
+                     user: UserModel,
+                     subject: String,
+                     textBody: String,
+                     htmlBody: String): Unit = {
+
+    val templateId: String = sys.env.get(templateIdKey) match {
+      case Some(id) => id
+      case _ => throw new NotificationClientException("Missing GovUk Notify templated id")
```

Comment:
Should our code be throwing `NotificationClientException`? Or is that an internal exception that only Notify code should be throwing?

Reply:
Good point